src/nvim/api/extmark.c

```diff
@@ -540,36 +541,29 @@ Integer nvim_buf_set_extmark(Buffer buffer, Integer ns_id, Integer line, Integer
     col2 = (int)val;
   }
 
-  // uncrustify:off
+  if (opts->hl_group) {
+    hl.hl_id = (int)opts->hl_group;
+    has_hl = true;
+  }
 
-  // TODO(bfredl): keyset type alias for hl_group? (nil|int|string)
-  struct {
-    const char *name;
-    Object *opt;
-    int *dest;
-  } hls[] = {
-    { "hl_group"           , &opts->hl_group           , &hl.hl_id            },
-    { "sign_hl_group"      , &opts->sign_hl_group      , &sign.hl_id       },
-    { "number_hl_group"    , &opts->number_hl_group    , &sign.number_hl_id     },
-    { "line_hl_group"      , &opts->line_hl_group      , &sign.line_hl_id       },
-    { "cursorline_hl_group", &opts->cursorline_hl_group, &sign.cursorline_hl_id },
-    { NULL, NULL, NULL },
-  };
+  if (opts->sign_hl_group) {
+    sign.hl_id = (int)opts->sign_hl_group;
```

Comment:
don't need these assignments be in if-statements. then we could just use `opts->sign_hl_group || opts->cursorline_hl_group || ...` in the end for the `sign.flags` check.